components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/JDBCConfigPersistenceManagerTest.java

```diff
@@ -0,0 +1,101 @@
+
+import static org.junit.Assert.assertFalse;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.wso2.carbon.identity.entitlement.PDPConstants.Algorithms.*;
+
+/**
+ * This class tests the behavior of the JDBCConfigPersistenceManager class.
+ */
+@WithCarbonHome
+@WithRegistry
+@WithRealmService(injectToSingletons = {EntitlementConfigHolder.class}, initUserStoreManager = true)
+@WithH2Database(jndiName = "jdbc/WSO2IdentityDB", files = {"dbscripts/h2.sql"})
+public class JDBCConfigPersistenceManagerTest {
+
+    private JDBCConfigPersistenceManager jdbcConfigPersistenceManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        jdbcConfigPersistenceManager = new JDBCConfigPersistenceManager();
+    }
+
+    @Test
+    public void testGetDefaultGlobalPolicyAlgorithmName() {
+
+        String globalPolicyAlgorithmName = jdbcConfigPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(globalPolicyAlgorithmName, DENY_OVERRIDES);
+    }
+
+    @Test(dependsOnMethods = {"testGetDefaultGlobalPolicyAlgorithmName"})
+    public void testAddGlobalPolicyAlgorithm() throws Exception {
+
+        // Add the first global policy combining algorithm.
+        boolean addResult = jdbcConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(FIRST_APPLICABLE);
+        // When adding the global policy combining algorithm for the first time, the result should be true.
+        assertFalse(addResult);
+
+        String policyAlgorithmNameFromDb = jdbcConfigPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(policyAlgorithmNameFromDb, FIRST_APPLICABLE);
+    }
+
+    @DataProvider
+    public Object[][] globalPolicyAlgorithmData() {
+
+        return new Object[][]{
+                {DENY_OVERRIDES},
+                {PERMIT_OVERRIDES},
+                {FIRST_APPLICABLE},
+                {ONLY_ONE_APPLICABLE},
+                {ORDERED_DENY_OVERRIDES},
+                {ORDERED_PERMIT_OVERRIDES}
+        };
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData", dependsOnMethods = {"testAddGlobalPolicyAlgorithm"})
+    public void testUpdateGlobalPolicyAlgorithm(String policyCombiningAlgorithm) throws Exception {
+
+        // Update the global policy combining algorithm.
+        boolean updateResult = jdbcConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyCombiningAlgorithm);
+        // When updating the global policy combining algorithm, the result should be true.
+        assertTrue(updateResult);
+
+        String policyAlgorithmNameFromDb = jdbcConfigPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(policyAlgorithmNameFromDb, policyCombiningAlgorithm);
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData", dependsOnMethods = {"testUpdateGlobalPolicyAlgorithm"})
+    public void testGetGlobalPolicyAlgorithm(String policyCombiningAlgorithm) throws Exception {
```

Comment:
What exactly this test validate that is not covered by the other tests up to this point?

Reply:
Yes. this is is not needed since the same scenarios were already covered through the previous test cases. So Fixed this in the latest changes.